hw/ip/aes/pre_sca/alma/cpp/testbench.h

```diff
@@ -0,0 +1,73 @@
+// Copyright IAIK.
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "verilated.h"
+#include "verilated_vcd_c.h"
+
+template <class Module>
+struct Testbench {
+  unsigned long m_tickcount;
+  Module *m_core;
+  VerilatedVcdC *m_trace;
+
+  Testbench(void) {
+    Verilated::traceEverOn(true);
+    m_core = new Module;
+    m_tickcount = 0ul;
+  }
+
+  virtual ~Testbench(void) {
+    delete m_core;
+    m_core = NULL;
+  }
+
+  virtual void opentrace(const char *vcdname) {
+    if (!m_trace) {
+      m_trace = new VerilatedVcdC;
+      m_core->trace(m_trace, 99);
+      m_trace->open(vcdname);
+    }
+  }
+
+  virtual void closetrace(void) {
```

Comment:
Should we also be calling this from the destructor?

Reply:
We call it inside the actual testbench but I agree it would make sense to call it in the destructor, too.